### DISCUSSION TITLE
Invalid PFT in history file with PFT 9

### DISCUSSION THREAD

**Discussion by Aariq** [EDmodel/ED2#371] [Q&A] [answered]
I'm getting an error about an invalid PFT in history file that I can't make sense of.  The only PFT in the history file is PFT 9 and it is also the only PFT set for  `NL%INCLUDE_THESE_PFT`.  I'm using IED_INIT_MODE=6.  I'm not sure where to look next to debug this.

```
Using cohort file: /data/sites/mandifore/FIA_ED2_site_1-21575/siteid1000021575.fia2000.radius0.075.lat44.1098lon-120.994.css
I found a cohort with PFT=     0  and it is not in your include_these_pft...
::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
 
--------------------------------------------------------------
                     !!! FATAL ERROR !!!                      
--------------------------------------------------------------
    ---> File:        ed_history_io.f90
    ---> Subroutine:  read_ed10_ed20_history_file
    ---> Reason:      Invalid PFT in history file
--------------------------------------------------------------

```

<details>
<summary> .css file </summary>

```
time patch cohort pft dbh hite n bdead balive lai
1999 50252 14 9 8.636 0 0.0104093346 0 0 0
1999 50252 24 9 7.239 0 0.0178445736 0 0 0
1999 50252 34 9 12.192 0 0.007435239 0 0 0
1999 50252 44 9 11.176 0 0.0118963824 0 0 0
1999 50288 14 9 26.416 0 0.006194241025 0 0 0
1999 50288 24 9 52.07 0 0.00197914745 0 0 0
1999 50288 34 9 65.786 0 0.0022252588029 0 0 0
1999 50288 44 9 30.48 0 0.004707193225 0 0 0
```

</details>

**Comment by mpaiao** [comment, marked as answer]
@Aariq note that the columns in your css file are not in the same order as shown in [the Wiki](https://github.com/EDmodel/ED2/wiki/Initial-conditions#files-types-and-formats-for-nlied_init_mode6). The header in the pss and css files are for reference only. The code does not read the header, it expects the data to be provided in the order shown in the table. 

This was not clear in [the Wiki](https://github.com/EDmodel/ED2/wiki/Initial-conditions#files-types-and-formats-for-nlied_init_mode6), so I just revised it to include a note about the order of the columns.

**Reply by Aariq** [reply]
🤦‍♂️Thanks, I accidentally re-ordered them.  I'll fix that and try again.

**Reply by Aariq** [reply]
Yup, that was my issue.  Thanks!
